pkg/cmd/flags/policy.go

```diff
@@ -5,114 +5,351 @@ import (
+				if err != nil {
+					return nil, err
+				}
+				continue
+			}
 
-				// retval
-				if strings.HasPrefix(f, "retval") {
-					filterFlags = append(filterFlags, &filterFlag{
-						full:              fmt.Sprintf("%s.%s", r.Event, f),
-						filterName:        fmt.Sprintf("%s.%s", r.Event, filterName),
-						operatorAndValues: operatorAndValues,
-					})
+			if scopeFlag.scopeName == "mntns" {
+				if strings.ContainsAny(scopeFlag.operator, "<>") {
+					return nil, filters.InvalidExpression(scopeFlag.operatorAndValues)
+				}
+				err := p.MntNSFilter.Parse(scopeFlag.operatorAndValues)
+				if err != nil {
+					return nil, err
+				}
+				continue
+			}
+
+			if scopeFlag.scopeName == "pidns" {
+				if strings.ContainsAny(scopeFlag.operator, "<>") {
+					return nil, filters.InvalidExpression(scopeFlag.operatorAndValues)
+				}
+				err := p.PidNSFilter.Parse(scopeFlag.operatorAndValues)
+				if err != nil {
+					return nil, err
+				}
+				continue
+			}
+
+			if scopeFlag.scopeName == "tree" {
+				err := p.ProcessTreeFilter.Parse(scopeFlag.operatorAndValues)
+				if err != nil {
+					return nil, err
+				}
+				continue
+			}
+
+			if strings.HasPrefix("pid", scopeFlag.scopeName) {
+				if scopeFlag.operatorAndValues == "=new" {
+					if err := p.NewPidFilter.Parse("new"); err != nil {
+						return nil, err
+					}
+					continue
+				}
+				if scopeFlag.operatorAndValues == "!=new" {
+					if err := p.NewPidFilter.Parse("!new"); err != nil {
+						return nil, err
+					}
 					continue
 				}
+				err := p.PIDFilter.Parse(scopeFlag.operatorAndValues)
+				if err != nil {
+					return nil, err
+				}
+				continue
+			}
+
+			if scopeFlag.scopeName == "uts" {
+				err := p.UTSFilter.Parse(scopeFlag.operatorAndValues)
+				if err != nil {
+					return nil, err
+				}
+				continue
+			}
+
+			if strings.HasPrefix("uid", scopeFlag.scopeName) {
+				err := p.UIDFilter.Parse(scopeFlag.operatorAndValues)
+				if err != nil {
+					return nil, err
+				}
+				continue
+			}
 
-				// context
-				filterFlags = append(filterFlags, &filterFlag{
-					full:              fmt.Sprintf("%s.context.%s", r.Event, f),
-					filterName:        fmt.Sprintf("%s.context.%s", r.Event, filterName),
-					operatorAndValues: operatorAndValues,
-				})
+			if strings.HasPrefix("follow", scopeFlag.scopeName) {
+				p.Follow = true
+				continue
 			}
+
+			return nil, InvalidScopeOptionError(scopeFlag.full, newBinary)
+		}
+
+		eventFilter := eventFilter{
+			Equal:    []string{},
+			NotEqual: []string{},
+		}
+
+		policyEvents, ok := policyEventsMap[policyIdx]
+		if !ok {
+			return nil, InvalidFlagEmpty()
+		}
+
+		for _, evtFlag := range policyEvents.eventFlags {
+			if evtFlag.operator == "" && evtFlag.eventOptionType == "" {
+				// no operator and no event option type means that the flag contains only event names
+				evtsNames := strings.Split(evtFlag.full, ",")
+				for _, evtName := range evtsNames {
+					if !strings.HasPrefix(evtName, "-") {
```

Comment:
We could change this in the future (other issue), considering "-" as a remove operator in the parse function, and setting the operator field accordingly. The same goes to scope "!" negation operator in scopes.

In other words. Parse those operators in their respective parse functions, parseScopeFlag and parseEventFlag, and checking the flag operator field when necessary in later stages. WDYT?

Reply:
https://github.com/aquasecurity/tracee/issues/3312